### DISCUSSION TITLE
Exposing internal load balancer for Google Cloud

### DISCUSSION THREAD

**Discussion by MandarJKulkarni** [strimzi/strimzi-kafka-operator#5176] [Q&A]
How can I expose internal load balancer for listener when deploying strimzi kafka cluster on GKE?

This blog post seems outdated.
https://strimzi.io/blog/2019/05/13/accessing-kafka-part-4/

**Comment by scholzj** [comment]
Well, it is more than 2 years old. But not sure what is outdated on it, you would need to be more specific. Did the GKE annotations changed for internal loadbalancers?

**Reply by MandarJKulkarni** [reply]
when I used below setting in the yaml, I got error that configuration is unrecognized field.

configuration:
      bootstrap:
        annotations:
          cloud.google.com/load-balancer-type: "Internal"

**Reply by scholzj** [reply]
Maybe you need to use the other option?

**Reply by MandarJKulkarni** [reply]
umm, you mean the template.extenalBootstrapService one?

I tried that too.

**Reply by MandarJKulkarni** [reply]
where can I find all the supported configurations for kafka cluster

**Reply by scholzj** [reply]
Well, I guess you will need to investigate it. First step would be to find out what operator version are you using and which configuration should you use. You can also doublecheck the example from the blog post against the documentation for your operator version. Also, to debug something like this, you should probably check your operator logs for any errors and check the services if they have the right annotation or not. Another good practice is to share your complete Kafka resource - not just some snippet. That way it is better to see if you placed it right etc.

**Reply by MandarJKulkarni** [reply]
Got it to work!
Here is the configuration that worked and created 2 internal load balancers.

`
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    version: 2.7.0
    replicas: 1
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
      - name: external
        port: 9094
        type: loadbalancer
        tls: false
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      log.message.format.version: "2.7"
      inter.broker.protocol.version: "2.7"
    template:
      externalBootstrapService:
        metadata:
          annotations:
            cloud.google.com/load-balancer-type: "Internal"
      perPodService:
        metadata:
          annotations:
            cloud.google.com/load-balancer-type: "Internal"
    storage:
      type: jbod
      volumes:
        - id: 0
          type: persistent-claim
          size: 100Gi
          deleteClaim: false
  zookeeper:
    replicas: 1
    storage:
      type: persistent-claim
      size: 100Gi
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}
`
